src/web/tools/KeyEventManager.ts

```diff
@@ -0,0 +1,81 @@
+import { AdaptedEvent, EventTypes } from '../interfaces';
+import EventManager from './EventManager';
+import { PointerType } from '../../PointerType';
+
+export default class KeyEventManager extends EventManager<HTMLElement> {
+  private activationKeys = ['Enter', ' '];
+  private cancelationKeys = ['Tab'];
+  private isPressed = false;
+
+  private keyDownCallback = (event: KeyboardEvent): void => {
+    if (this.cancelationKeys.indexOf(event.key) !== -1 && this.isPressed) {
+      this.dispatchEvent(event, EventTypes.CANCEL);
+      return;
+    }
+
+    if (this.activationKeys.indexOf(event.key) === -1) {
+      return;
+    }
+
+    this.dispatchEvent(event, EventTypes.DOWN);
+  };
+
+  private keyUpCallback = (event: KeyboardEvent): void => {
+    if (this.activationKeys.indexOf(event.key) === -1 || !this.isPressed) {
+      return;
+    }
+
+    this.dispatchEvent(event, EventTypes.UP);
+  };
+
+  private dispatchEvent(event: KeyboardEvent, eventType: EventTypes) {
+    const adaptedEvent = this.mapEvent(event, eventType);
+    switch (eventType) {
+      case EventTypes.UP:
+        this.isPressed = false;
+        this.onPointerUp(adaptedEvent);
+        break;
+      case EventTypes.DOWN:
+        this.isPressed = true;
+        this.onPointerDown(adaptedEvent);
+        break;
+      case EventTypes.CANCEL:
+        this.isPressed = false;
+        this.onPointerCancel(adaptedEvent);
+        break;
+    }
+  }
+
+  public registerListeners(): void {
+    this.view.addEventListener('keydown', this.keyDownCallback);
+    this.view.addEventListener('keyup', this.keyUpCallback);
+  }
+
+  public unregisterListeners(): void {
+    this.view.addEventListener('keydown', this.keyDownCallback);
+    this.view.addEventListener('keyup', this.keyUpCallback);
+  }
+
+  protected mapEvent(
+    event: KeyboardEvent,
+    eventType: EventTypes
+  ): AdaptedEvent {
+    const viewRect = (event.target as HTMLElement)?.getBoundingClientRect?.();
```

Comment:
Is `?.` necessary after `getBoundingClientRect`?

Reply:
Yes, because `event.target` may not always be `HTMLElement` as described by [this documentation](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget).
I've checked if that occurs, and I've actually had that happen to me a couple of time during loading, which without `?.` resulted in a wall of red errors (although these were isolated and no crashes occured).

Reply:
Since we moved from `measure` to `getBoundingClientRect`, i'll add an additional null check to ensure no malformed events are sent.

Reply:
Added updated null-check as of [0a5855f](https://github.com/software-mansion/react-native-gesture-handler/pull/3035/commits/0a5855fa4cd9450aa41ff5bdc5999c06e06db4da)